src/contacts/Dockerfile

```diff
@@ -12,18 +12,30 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM python:3.8
- 
+# The Google App Engine python runtime is Debian Jessie with Python installed
+# and various os-level packages to allow installation of popular Python
+# libraries. The source is on github at:
+#   https://github.com/GoogleCloudPlatform/python-docker
+FROM gcr.io/google-appengine/python
+
 # show python logs as they occur
 ENV PYTHONUNBUFFERED=0
 
-# get packages
-WORKDIR /contacts
-COPY requirements.txt .
-RUN pip install --trusted-host pypi.python.org -r requirements.txt
+# Create a virtualenv for the application dependencies.
+RUN virtualenv -p python3.7 /env
```

Comment:
Is there a particular reason why we are using venv, usually I have only seen this in the context of local dev.
Or are we evaluating multi stage builds for leaner images?

Reply:
I'm following the pattern [recommended in the repo](https://github.com/GoogleCloudPlatform/python-runtime#kubernetes-engine--other-docker-hosts) for the official Google python image

I'd guess the reason is to make sure that all dependencies are captured in requirements.txt and we're not accidently relying on the system packages for anything, as those may shift over time

Reply:
oh makes sense